pages/api/trip/[tripID]/event/[eventID]/index.ts

```diff
@@ -0,0 +1,132 @@
+import { NextApiRequest, NextApiResponse } from "next"
+import { firebaseAuth, unpackArrayResponse } from "../../../../../../utility/firebase"
+import firebaseAdmin from "../../../../../../utility/firebaseAdmin"
+
+function findEventIndex(intervals: any, eventID: string) {
+  let i
+  for (i = 0; i < intervals.length; i += 1) {
+    const [start, end, uid] = intervals[i]
+    if (uid === eventID) break
+  }
+  if (i === intervals.length) return -1
+  return i
+}
+
+function mergeIntervals(intervals: any, eventID: string) {
+  const indexToRemove: number = findEventIndex(intervals, eventID)
+  if (indexToRemove === -1) {
+    return -1
+  }
+
+  let result = [[intervals[0][0], intervals[0][1]]]
+  const length: number = intervals.length
+  for (let i = 1; i < length; i += 1) {
+    if (intervals[i][0] <= result[result.length - 1][1]) {
+      result[result.length - 1][1] = Math.max(result[result.length - 1][1], intervals[i][1])
+    } else {
+      result.push(intervals[i])
+    }
+  }
+  return result.length
+}
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
```

Comment:
Instead of taking in req, res you can make this a function that can be called in trip/[tripID]/event by adding a callback that will do the whole res.status stuff if successful. So export default async function
```suggestion
 export default function maybeReschedule(newInterval: Duration, callback(isSuccess:boolean, errorMessage?:string) => void){
 
 }
```